src/bt_peer_connection.cpp

```diff
@@ -121,6 +121,35 @@ namespace {
 } // anonymous namespace
 #endif
 
+bool validate_hash_request(hash_request const& hr, file_storage const& fs)
+{
+	if (hr.file < file_index_t{0}
+		|| hr.file >= fs.end_file()
+		|| hr.base < 0
+		|| hr.index < 0
+		|| hr.count <= 0
+		|| hr.count > 8192
+		|| hr.proof_layers < 0)
+		return false;
+
+	int const num_leafs = merkle_num_leafs(fs.file_num_pieces(hr.file));
```

Comment:
for instance, I just look at the size of the file here, regardless of whether we have those hashes.

Reply:
I did that at first, but I was thrown off by the function `merkle_num_leafs()`, its parameter is called `pieces`. Maybe because in the old merkle implementation the pieces were the leaves.